runtime/lua/vim/lsp/_folding_range.lua

```diff
@@ -83,6 +87,9 @@ end
 --- Force `foldexpr()` to be re-evaluated, without opening folds.
 ---@param bufnr integer
 local function foldupdate(bufnr)
+  if not api.nvim_buf_is_loaded(bufnr) or not vim.b[bufnr].__lsp_folding_range_enabled then
```

Comment:
double-underscore is typically understood as Lua-internal things. single-underscore should be enough for a signal for Nvim-internal things.

I'm not sure that an underscore is needed for `b:` variables though, because that is not a "public" namespace, it's understood that things there are owned by the prefix family (`lsp` in this case).

```suggestion
  if not api.nvim_buf_is_loaded(bufnr) or not vim.b[bufnr]._lsp_folding_range_enabled then
```

Reply:
Got it, I added these underscores mainly to indicate that it is not a supported API. I will apply the single-undercore way.